src/Features/CSharp/Portable/Completion/CompletionProviders/AwaitCompletionProvider.cs

```diff
@@ -50,6 +48,23 @@ private protected override int GetSpanStart(SyntaxNode declaration)
         }
 
         private protected override SyntaxNode? GetAsyncSupportingDeclaration(SyntaxToken token)
-            => token.GetAncestor(node => node.IsAsyncSupportingFunctionSyntax());
+        {
+            var node = token.GetAncestor(node => node.IsAsyncSupportingFunctionSyntax());
+            // For local functions, make sure we either have arrow token or open brace token.
+            // Consider the following scenario:
+            // void Method()
+            // {
+            //     aw$$ AnotherMethodCall(); // NOTE: Here, the compiler produces LocalFunctionStatementSyntax.
+            // }
+            // For this case, we're interested in putting async in front of Method()
+            if (node is not LocalFunctionStatementSyntax localFunction ||
+                localFunction.ExpressionBody is { ArrowToken.RawKind: not 0 } ||
+                localFunction.Body is { OpenBraceToken.RawKind: not 0 })
+            {
+                return node;
+            }
+
+            return node.Parent?.FirstAncestorOrSelf<SyntaxNode>(node => node.IsAsyncSupportingFunctionSyntax());
```

Comment:
This is hard to parse, because the comment talks about the exception case, but that case is actually handled by the return at the end of the method. Couldn't this be re-written as:

```suggestion
            if (node is LocalFunctionStatementSyntax { ExpressionBody: null, Body: null })
            {
                return node.Parent?.FirstAncestorOrSelf<SyntaxNode>(node => node.IsAsyncSupportingFunctionSyntax());
            }

            return node;
```

Reply:
I think it *should* work, and it's much much better!